synapse/rest/client/register.py

```diff
@@ -564,6 +574,7 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
                 desired_username,
                 guest_access_token=guest_access_token,
                 assigned_user_id=registered_user_id,
+                inhibit_user_in_use_error=self._inhibit_user_in_use_error,
```

Comment:
I'm not adding another check later on (e.g. after we've checked that the UIA flow is complete) because the registration handler's `register_user` already does that.